internal/handlers/pg/msg_count.go

```diff
@@ -85,37 +88,46 @@ func (h *Handler) MsgCount(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, e
 
 	resDocs := make([]*types.Document, 0, 16)
 	err = h.PgPool.InTransaction(ctx, func(tx pgx.Tx) error {
-		fetchedChan, err := h.PgPool.QueryDocuments(ctx, tx, &sp)
-		if err != nil {
-			return err
+		it, ierr := h.PgPool.GetDocuments(ctx, tx, &sp)
```

Comment:
```suggestion
		it, err := h.PgPool.GetDocuments(ctx, tx, &sp)
```

Otherwise, we will have to very careful with both `err` and `ierr` around

Reply:
@AlekSi that's true, but govet complains about shadowing if I use `err`. Should we adjust something for `govet`?

Reply:
The other option would be something like this, I think:
```suggestion
		var it iterator.Interface[uint32, *types.Document]
		it, err = h.PgPool.GetDocuments(ctx, tx, &sp)
```